src/olympia/addons/models.py

```diff
@@ -510,15 +511,20 @@ def initialize_addon_from_upload(cls, data, is_listed=True):
         if not locale_is_set:
             addon.default_locale = to_language(trans_real.get_language())
 
-        addon.save()
+        if data.get('is_webextension', False):
+            # Resolve correct translations for WebExtensions first and save
+            cls.resolve_webext_translations_and_save(addon, data, upload)
```

Comment:
me neither but see the comment there for details. Otherwise I could set the name here and update the other things in `resolve_webext_translations` or so, maybe that's a bit clearer.